src/diagnostics/logPointsClient.ts

```diff
@@ -93,57 +116,75 @@ export class KuduLogPointsDebuggerClient extends LogPointsDebuggerClientBase imp
+        const opts = {
+            uri: `${KuduLogPointsDebuggerClient.getBaseUri(siteName)}/api/command`,
+            auth: KuduLogPointsDebuggerClient.getAuth(publishCredential),
+            json: true,
+            body: {
+                command: `/usr/bin/node ./ssh-client.js --command ${encodedCommand}`,
+                dir: 'site/wwwroot'
+            }
         };
 
-        const r = req.defaults({
-            baseUrl: `https://${siteName}.scm.azurewebsites.net/`,
-            headers: headers
-        });
+        const request = req.defaults({});
+        request.cookie(`ARRAffinity=${affinityValue}`);
 
-        r.cookie(`ARRAffinity=${affinityValue}`);
-        // tslint:disable-next-line:no-any
-        let cb: (err: any, body: any, response: any) => void;
-        const promise = new Promise<CommandRunResult<ResponseType>>((resolve, reject) => {
-            cb = (err, body?, response?) => {
+        return new Promise<CommandRunResult<ResponseType>>((resolve, reject) => {
+            request.post(opts, (err, httpResponse, body) => {
+                this.log(`sendCommand(): response is ${httpResponse.statusCode}`);
                 if (err) {
+                    this.log(`sendCommand(): received error: ${err}`);
                     reject(err);
-                    return;
-                }
-
-                if (response.statusCode !== 200) {
-                    reject(new Error(`${response.statusCode}: ${body}`));
-                    return;
+                } else if (httpResponse.statusCode === 200) {
+                    this.log(`sendCommand():  body is ${body}`);
+                    resolve(new CommandRunResult<ResponseType>(body.Error, body.ExitCode, body.Output));
+                } else {
+                    reject(`${httpResponse.statusCode} - ${httpResponse.statusMessage}`);
                 }
-
-                resolve(new CommandRunResult<ResponseType>(body.Error, body.ExitCode, body.Output));
-            };
+            });
         });
+    }
 
-        r.post(
-            {
-                uri: "/api/command",
-                json: {
-                    command: command,
-                    dir: '/'
-                }
+    private log(message: string): void {
+        // tslint:disable-next-line:no-unused-expression
+        message && 1;
+    }
+
+    private async uploadSshClient(siteName: string, affinityValue: string, publishCredential: WebSiteModels.User): Promise<void> {
+        const request = req.defaults({});
+        request.cookie(`ARRAffinity=${affinityValue}`);
+        const opts = {
+            uri: `${KuduLogPointsDebuggerClient.getBaseUri(siteName)}/api/zip/site/wwwroot/`,
+            headers: {
+                'Content-Type': 'multipart/form-data'
             },
-            (err, response, body) => {
+            auth: KuduLogPointsDebuggerClient.getAuth(publishCredential),
+            body: fs.createReadStream(path.join(__dirname, '../../../resources/ssh-client.zip'))
```

Comment:
I'm a bit concerned here:

1. A Zip file is not as easy to track with Git. Is there a specific reason to send the files as a Zip, like upload performance or something?
2. What if the customer's website happens to have a file in the root folder also called ssh-client.js (or ssh-client.js.map)?

Reply:
1. For both convenience and performance considerations. At this point, we don't know how many files we need to deploy to Kudu container. Potentially we will deploy a `node_moudules` set of files as well. The zip is created in another repository, so we have the content versioned in that repository already. I can change it to uploading one file today. But it is likely to change to uploading a zip again in the future anyway. Optionally we can webpack everything together into one file. But the content of the end result will not make sense for the context of an extension code base. So it acts more like just a binary.

This is supposed to be a utility that once tested, will not change frequently.

2. That's a valid point. I can optionally put it to "/home/logpoints_sshclient". How does that sound?

Reply:
I'd suggest we have a directory structure like this:
  - `/home/logpoints/ssh-client`   // for ssh-client
  - `/home/logpoints/logs`           
  - `/home/logpoints/logs/agent`  //for logpoint-agent logs

Reply:
PTAL

Reply:
PTAL

Reply:
Thanks for the explanation. I understand the ssh-client is versioned in a separate repository. However, this still doesn't solve the problem in **this** repository that there is no easy way to know the version of the current ssh-client.zip, nor can you tell which version of the ssh-client.zip it was from a commit history. Maybe adding a txt file describing the version of the zip file? Anyways, this is not something worth blocking the merge.